jo-rdan/prop-lite
# implement the signup feature with db

#### What does this PR do?
It implements the signup endpoint using persistence

#### Description of Task to be completed
- It saves user data entered in the request in the database using SQL queries.
- It assigns token to the newly created user and encrypts the password

#### How should this be manually tested?
- First make sure the tables are created in the database if they're not run this command: **npm run db** to create tables 
- Start the the server by typing in this command in the terminal : **npm start**
- In POSTMAN enter required fields, If they are all valid it returns the created user with a 201 HTTP status code. Otherwise, it returns relevant error message.

#### Any background context you want to provide?
-N/A

#### What are the relevant pivotal tracker stories?
[#167281049](https://www.pivotaltracker.com/story/show/167281049)
